src/SmartComponents/ActivityTable/Columns.js

```diff
@@ -17,25 +22,87 @@ TaskNameCell.propTypes = {
   index: propTypes.number,
 };
 
-const StatusCell = ({ status }) =>
-  ({
-    Completed: (
-      <span style={{ color: '#3E8635', display: 'flex', alignItems: 'center' }}>
-        <Icon status="success" style={{ marginRight: '4px' }}>
-          <CheckCircleIcon />
-        </Icon>
-        {status}
-      </span>
-    ),
-    Running: (
-      <span style={{ color: '#2B9AF3', display: 'flex', alignItems: 'center' }}>
-        <Icon status="info" style={{ marginRight: '4px' }}>
-          <InProgressIcon />
-        </Icon>
-        {status}
-      </span>
-    ),
-  }[status] || status);
+const getStatusProps = (status) => {
+  let badgeColor, iconStatus, IconType, message;
+
+  switch (status) {
+    case TASK_STATUS.RUNNING:
+      badgeColor = '#2B9AF3';
+      iconStatus = 'info';
+      IconType = InProgressIcon;
+      message = 'This task still has jobs in the running state';
+      break;
+    case TASK_STATUS.COMPLETED:
+      badgeColor = '#3E8635';
+      iconStatus = 'success';
+      IconType = CheckCircleIcon;
+      message = 'All jobs in this task have completed successfully';
+      break;
+    case TASK_STATUS.COMPLETED_WITH_ERRORS:
+      badgeColor = '#FF0000';
+      iconStatus = 'danger';
+      IconType = ExclamationCircleIcon;
+      message = 'Some jobs in this task have failed to complete successfully';
+      break;
+    case TASK_STATUS.FAILURE:
+      badgeColor = '#FF0000';
+      iconStatus = 'danger';
+      IconType = ExclamationCircleIcon;
+      message = 'All jobs in this task have failed to complete successfully';
+      break;
+    case TASK_STATUS.CANCELLED:
+      badgeColor = '#FF0000';
+      iconStatus = 'danger';
+      IconType = ExclamationCircleIcon;
+      message = 'All jobs in this task have been cancelled';
+      break;
+  }
```

Comment:
I spoke with Andy Burka, and he needs to define these `message`s for each of the statuses. For the most part, they will be "canned"; but a task with a failure will need to use a message returned from the backend that gives more details of the failure.

Reply:
Do we want to try to include the number of jobs that have failed / are still running / have completed?

Reply:
The message for a failed task may be different for each job that ran as part of the task.  Which one do we pick?  Also, we might not really know why a job failed other than the last thing we logged about it before it failed.  We don't get good error reporting back from Playbook Dispatcher.  Perhaps a canned message might be enough to let the user know they need to drill into each failed job to find out why each failed?

Reply:
I want to make sure we aren't confusing Failed with Completed with errors. If I understand correctly, Failed means that the task failed to run on any system. Meaning that it failed at the highest level. I would assume this means that the tasks backend failed to run the script, or the script failed before it ever communicated with a system. I'm not sure if this is even possible, so any feedback is appreciated.

Let me know what you think about this ^